contrib/systemd/master/atomic-openshift-master-api

```diff
@@ -0,0 +1,64 @@
+#! /bin/bash
+
+set -euo pipefail
+
+pods_dir=/etc/origin/node/pods
+pods_disabled_dir=/etc/origin/node/pods.disabled
+
+static_pod_file=apiserver.yaml
+
+# Make sure both directories exist
+mkdir -p $pods_dir $pods_disabled_dir
+
+function get_container_values() {
+    child_container=$(docker ps -q --filter "label=io.kubernetes.container.name=api")
+    container=$(docker ps -q --filter "label=openshift.io/component=api" --filter "label=io.kubernetes.container.name=POD")
+}
+
+
+# See how we were called.
+case "$1" in
+start)
+    # See if api server is started
+    if [ -f "${pods_dir}/${static_pod_file}" ]; then
+        echo "api server already started"
+        exit 0
+    fi
+
+    # Start the api server pod
+    mv "${pods_disabled_dir}/${static_pod_file}" "${pods_dir}"
+
+    sleep 1 # avoid race condition
+
+    # Wait for the pod to come up
+    while ! docker ps -q --filter "label=io.kubernetes.container.name=api" &> /dev/null ; do
+        sleep 1
+    done
+
+    get_container_values
+    docker logs --follow "$child_container"
+    exit $?
+    ;;
+
+stop)
+    # See if api server is started
+    if [ -f "${pods_disabled_dir}/${static_pod_file}" ]; then
+        echo "api server already stopped"
+        exit 0
+    fi
+   
+    get_container_values
+
+    # Start the api server pod
+    mv "${pods_dir}/${static_pod_file}" "${pods_disabled_dir}"
+
+    exec timeout 60 docker wait $child_container
```

Comment:
In the demo, we saw that sometimes the timeout hits before the container is fully stopped.

I'm fine with this because the `mv` above means that the kubelet will eventually stop the pod, but @derekwaynecarr had expressed concern with this and possibly wanting a better solution.

@derekwaynecarr is this sufficient or do you still want that?

Reply:
it looks like we wait 60s here, but the unit is configured to wait 10m.

the systemd default timeout appears to be 90s.

i am fine if we set 10m, but we should then make the two timeouts the same value (i.e. 10m or 600s).

Reply:
@derekwaynecarr @sjenning  Ok, if we're just going to change the timeout, then be aware that saw an odd behavior where even when docker said the child_container was no longer running, the docker wait didn't return (it ended up timing out).

I didn't mention it before because I was under the impression that the `docker wait` was going away.

Could we test to make sure that if we set the timeout to 10 minutes that when the docker container has exited, the `docker wait` actually complete?

I'm worried that in many cases a docker restart (stop / start) will always take 10 minutes.

Thanks!

Reply:
@derekwaynecarr - Thanks for the review. I have changed the time to 600 sec in script. 

@twiest - I have tested the scenario, for example 
- I started a nginx container `docker run nginx`  with <container_id> 
- Run `exec 600 docker wait <container_id>`. 
- I killed the container using docker kill `<container_id>`.
- The exec returned immediately.

Is this the same thing that you wanted to test?

Reply:
@ravisantoshgudimetla Did you mean `exec timeout 600 docker wait <container_id>` ?

If so, then yes.

Basically just use the `systemctl start` and then `systemctl stop` and if it stops when the container stops, then we're good.  If not, then we may need something else.

Thanks!

Reply:
> Did you mean exec timeout 600 docker wait <container_id> ?

Yes.

Thanks !!